### PR TITLE
[dhcp6relay] Add dhcpv6 option check

### PR DESCRIPTION
**Pull request by kellyyeh**
#### Why I did it
To prevent relaying client packets with malformed payload

#### How I did it
Check for invalid dhcpv6 options

#### Which release branch to backport (provide reason below if selected)

- [x] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106
- [x] 202111

### CODE REVIEW

File: src/dhcp6relay/src/relay.cpp [hunk 1]
```diff
@@ -609,21 +609,44 @@ void callback(evutil_socket_t fd, short event, void *arg) {
...
+
     counters[msg->msg_type]++;
     std::string counterVlan = counter_table;
     update_counter(config->db, counterVlan.append(config->interface), msg->msg_type);
 
+    if(msg->msg_type != DHCPv6_MESSAGE_TYPE_RELAY_FORW) {
+        while (option_position - (uint8_t *)ptr < len) {
+            auto option = parse_dhcpv6_opt(option_position, &tmp);
+            option_position = tmp;
+            if(ntohs(option->option_code) > 56) {   // DHCPv6 option code greater than 56 are currently unassigned
```
Comment (yxieca) [inline src/dhcp6relay/src/relay.cpp:622]: Define a macro/constant for 56

**Comment by yxieca** [comment]
Add unit test?

**Comment by kellyyeh** [comment]
> Add unit test?

Noted

File: src/dhcp6relay/src/relay.cpp (hunk 1 above)
Comment (kellyyeh) [reply src/dhcp6relay/src/relay.cpp]: Updated

File: src/dhcp6relay/src/relay.cpp [hunk 2]
```diff
@@ -609,21 +609,44 @@ void callback(evutil_socket_t fd, short event, void *arg) {
...
     counters[msg->msg_type]++;
     std::string counterVlan = counter_table;
     update_counter(config->db, counterVlan.append(config->interface), msg->msg_type);
 
+    if(msg->msg_type != DHCPv6_MESSAGE_TYPE_RELAY_FORW) {
+        while (option_position - message_buffer < len) {
+            auto option = parse_dhcpv6_opt(option_position, &tmp);
+            option_position = tmp;
+            if(ntohs(option->option_code) > DHCPv6_OPTION_LIMIT) {
+                syslog(LOG_INFO, "DHCPv6 option is invalid or contains malformed payload\n");
```
Comment (yxieca) [inline src/dhcp6relay/src/relay.cpp:623]: We should raise the log level to warning and add a counter for this.

**Review by yxieca — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Comment by liuh-80** [comment]
202012 branch PR merged: https://github.com/sonic-net/sonic-buildimage/pull/10808

**Pull request merged**